### PR TITLE
Finetune section refactoring

### PR DESCRIPTION
**Pull request by JegernOUTT**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -45,6 +45,10 @@ ENV TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
 COPY . /tmp/app
 RUN pip install /tmp/app && rm -rf /tmp/app
 
+RUN git clone -b feat/alibi https://github.com/smallcloudai/flash-attention.git /tmp/flash-attention \
+    && cd /tmp/flash-attention \
+    && MAX_JOBS=8 python3 setup.py install
+
```
Comment (mitya52) [inline Dockerfile:51]: we should add flash_attn in setup, something like this
```
"flash_attn @ git+https://github.com/smallcloudai/flash-attention@feat/alibi"
```

to build it in docker you need some env vars:
```
ENV MAX_JOBS=8
ENV FLASH_ATTENTION_FORCE_BUILD="TRUE"
```

I tested this in docker image but not with docker build. Hope it'll help!

**Review by mitya52 — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (JegernOUTT) [reply Dockerfile:51]: So, have you succeeded in building the necessary version just with `flash_attn @ git+https://github.com/smallcloudai/flash-attention@feat/alibi` ? Or you're suggesting to keep them both in the docker file and `setup.py`?

File: Dockerfile [hunk 2]
```diff
@@ -45,6 +45,12 @@ ENV TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
 COPY . /tmp/app
 RUN pip install /tmp/app && rm -rf /tmp/app
 
+ENV MAX_JOBS=4
+ENV FLASH_ATTENTION_FORCE_BUILD="TRUE"
+RUN git clone -b feat/alibi https://github.com/smallcloudai/flash-attention.git /tmp/flash-attention \
+    && cd /tmp/flash-attention \
+    && python3 setup.py install
```
Comment (mitya52) [inline Dockerfile:52]: move it to setup, should work

Comment (mitya52) [reply Dockerfile:52]: in addition set env vars before refact package install

File: setup.py [hunk 3]
```diff
@@ -65,7 +69,21 @@ def find_required_packages(packages: Set[str]) -> Set[str]:
     return packages
 
 
-setup_package = os.environ.get("SETUP_PACKAGE", None)
+def get_install_requires():
```
Comment (mitya52) [inline setup.py:72]: pass setup_package as arg or something

**Review by mitya52 — CHANGES_REQUESTED** [review]

**Pull request merged**